TrackletAlgorithm/MatchProcessor.h

```diff
@@ -21,45 +21,63 @@ namespace PR
   // Initialization
   // check the number of entries in the input memories
   // fill the bit mask indicating if memories are empty or not
-  template<int nMEM, int NBits_Entries, class MemType>
+  template<int nMEM, unsigned int nINMEM, int NBits_Entries, class MemType>
   inline void init(BXType bx,
+		   const ap_uint<5> iMem[nMEM],
+		   const ap_uint<2> iPage[nMEM],
+		   const ap_uint<3> nPages[nINMEM],
```

Comment:
@aryd Since these three types are used in several places, could you please give them meaningful names with `typedef`, to remove the magic numbers?